pkg/blockdevice/new_block_device_from_file_windows.go

```diff
@@ -0,0 +1,95 @@
+//go:build windows
+// +build windows
+
+package blockdevice
+
+import (
+	"syscall"
+	"unsafe"
+
+	"github.com/buildbarn/bb-storage/pkg/util"
+
+	"golang.org/x/sys/windows"
+)
+
+var (
+	kernel32          = windows.NewLazySystemDLL("kernel32.dll")
+	GetDiskFreeSpaceW = kernel32.NewProc("GetDiskFreeSpaceW")
+)
+
+// NewBlockDeviceFromFile creates a BlockDevice that is backed by a
+// regular file stored in a file system.
+func NewBlockDeviceFromFile(path string, minimumSizeBytes int, zeroInitialize bool) (BlockDevice, int, int64, error) {
+	pathPtr, err := syscall.UTF16PtrFromString(path)
+	if err != nil {
+		return nil, 0, 0, util.StatusWrapf(err, "Failed to convert path %#v to UTF-16", path)
+	}
+
+	var createmode uint32 = windows.OPEN_ALWAYS
+	if zeroInitialize {
+		createmode = windows.CREATE_ALWAYS
+	}
+	handle, err := windows.CreateFile(
+		pathPtr,
+		windows.GENERIC_READ|windows.GENERIC_WRITE,
+		windows.FILE_SHARE_READ|windows.FILE_SHARE_WRITE,
+		nil,
+		createmode,
+		windows.FILE_ATTRIBUTE_NORMAL,
+		0)
+	if err != nil {
+		return nil, 0, 0, util.StatusWrapf(err, "Failed to open file %#v", path)
+	}
+
+	// Get the sector size from the disk where the file is located
+	var sectorSizeBytes int
+	if len(path) >= 3 && path[1] == ':' && path[2] == '\\' {
+		// Extract the root path (drive letter) from the full path
+		var rootPath string = path[:3]
+		rootPathPtr, err := syscall.UTF16PtrFromString(rootPath)
+		if err != nil {
+			windows.CloseHandle(handle)
+			return nil, 0, 0, util.StatusWrapf(err, "Failed to convert root path %#v to UTF-16", rootPath)
+		}
+
+		// Get the sector size using GetDiskFreeSpace via syscall since it's not in windows package
+		var bytesPerSector uint32
+		r, _, err := GetDiskFreeSpaceW.Call(
+			uintptr(unsafe.Pointer(rootPathPtr)),
+			0,
```

Comment:
To improve readability, could you please add some `/* sectorsPerCluster = */ 0,` comments in front of the fields that you're setting to zero? That way it becomes more evident why we do.